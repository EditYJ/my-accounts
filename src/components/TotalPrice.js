import React from "react";
import PropTypes from "prop-types";

const TotalPrice = ({ income, outcome }) => {
  return (
    <div className="row">
      <h5 className="col">
        收入: <span>{income}</span>
      </h5>
      <h5 className="col">
        支出: <span>{outcome}</span>
      </h5>
    </div>
  );
};

TotalPrice.protoType = {
  income: PropTypes.number.isRequired,
  outcome: PropTypes.number.isRequired
};

TotalPrice.defaultProps = {
  income: 0,
  outcome: 0
};

export default TotalPrice;
