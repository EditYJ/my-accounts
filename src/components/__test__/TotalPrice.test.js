import React from "react";
import { shallow } from "enzyme";
import TotalPrice from "../TotalPrice";

const props = {
  income: 1000,
  outcome: 2000
};

let wrapper;
describe("test TotalPrice component", () => {
  beforeEach(() => {
    wrapper = shallow(<TotalPrice {...props} />);
  });

  it("should render the component to match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("component should render correct income&outcome number", () => {
    expect(wrapper.find(".income span").text() * 1).toEqual(1000);
    expect(wrapper.find(".outcome span").text() * 1).toEqual(2000);
  });
});
