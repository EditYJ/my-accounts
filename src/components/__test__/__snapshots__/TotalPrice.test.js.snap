// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test TotalPrice component should render the component to match snapshot 1`] = `
<div
  className="row"
>
  <h5
    className="col income"
  >
    收入: 
    <span>
      1000
    </span>
  </h5>
  <h5
    className="col outcome"
  >
    支出: 
    <span>
      2000
    </span>
  </h5>
</div>
`;
